crossbar/worker/container.py

```diff
@@ -235,9 +237,15 @@ def start_container_component(self, id, config, reload_modules=False, details=No
             self._module_tracker.reload()
 
         # WAMP application session factory
-        #
+        # ultimately, this gets called once the connection is
+        # establised, from onOpen in autobahn/wamp/websocket.py:59
         def create_session():
-            return create_component(componentcfg)
+            try:
+                return create_component(componentcfg)
+            except Exception:
+                # AutobahnPython swallows exceptions from onOpen
+                log.err(_why="Instantiating component failed")
+                raise
 
```

Comment:
Instead of the "2 second" thing in node.py, we could instead call `create_component()` right away and have `create_session` be a closure that just returns the instance; then we could fail right away on instantiation problems. This would work for constructor problems, but not anything else.